addusers.py

```diff
@@ -201,7 +203,106 @@ def exists_in_keystone(check_thing, keystone_things):
         return None
 
 
-def parse_rows(rows, select_user=None):
+def remove_duplicates(rows, UC, PC, EC):
+    """Deletes all duplicate rows.
+    user_id_column: UC
+    new_project_column: PC
+    existing_project_column: EC
+    """
+    if len(rows) <= 1:
+        return rows
+    else:
+        output = []
+        for i in range(len(rows)):
+            flag = True
+            for j in rows[:i]:
+                if rows[i][UC] == j[UC] and\
+                   rows[i][PC] == j[PC] and\
+                   rows[i][EC] == j[EC]:
+                    flag = False
+            if flag:
+                output.append(rows[i])
+    return output
+
+
+def create_new_users_rows(rows):
+    output = []
+    for entry in rows:
+        if entry[7] == 'No':
+            user = [entry[4], entry[5], entry[6], entry[8],
+                    entry[9], entry[10], entry[11], entry[12],
+                    entry[13], entry[14], entry[15], entry[16],
+                    entry[18], entry[19], entry[20], entry[21]]
+            output.append(user)
+    return output
+
+
+def create_new_projects_rows(rows):
+    output = []
+    for entry in rows:
+        if entry[15] != "":
+            project = [entry[4], entry[5], entry[6], entry[8],
+                       entry[9], entry[10], entry[11], entry[12],
+                       entry[13], entry[16], entry[17]]
+            output.append(project)
+    return output
+
+
+def manual_testing(header):
+    """Testing changes to the code that doesn't involve the form
+    """
+    print "...TESTING..."
+    header_option = raw_input("Create own header?     [y/n] ")
+    if header_option == 'y':
+        header = []
+        while True:
+            idx = raw_input(("Enter a header_column. "
+                             "Leave empty if you're done: "))
+            if idx == '':
+                break
+            header.append(idx)
+    tests = []
+    tests.append(header)
+    num_tests = int(raw_input("Number of tests: "))
+    for idx in range(num_tests):
+        index_length = len(header)
+        index_flag = raw_input("New Project?   [y/n] ")
+        if index_flag == 'y':
+            index_length -= 4
+        entry = [0] * len(header)
+        for i in range(len(entry)):
+            entry[i] = raw_input(header[i] + ": ")
+        tests.append(entry)
+    return tests
+
+
+def manual_fix(rows):
```

Comment:
An incorrect input could be a typo that the user made.
For instance, the incorrect input is '1234@email.com', but the correct input is '5678@email.com'.